v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/utils/BigtableUtils.java

```diff
@@ -0,0 +1,305 @@
+        BigtableUtils.bigtableRowColumnFamilyName,
+        ChangelogColumns.COLUMN_FAMILY.getColumnNameAsByteBuffer(),
+        entry.getTimestamp(),
+        getByteBufferFromString(String.valueOf(entry.getTieBreaker()), charset)
+    ));
+
+    // low_watermark
+    cells.add(new com.google.cloud.teleport.bigtable.BigtableCell(
+        BigtableUtils.bigtableRowColumnFamilyName,
+        ChangelogColumns.LOW_WATERMARK.getColumnNameAsByteBuffer(),
+        entry.getTimestamp(),
+        getByteBufferFromString(String.valueOf(entry.getLowWatermark()), charset)
+    ));
+
+    if (entry.getColumn() != null) {
+      // column
+      cells.add(new com.google.cloud.teleport.bigtable.BigtableCell(
+          BigtableUtils.bigtableRowColumnFamilyName,
+          ChangelogColumns.LOW_WATERMARK.getColumnNameAsByteBuffer(),
+          entry.getTimestamp(),
+          getByteBufferFromString(String.valueOf(entry.getLowWatermark()), charset)
+      ));
+    }
+
+    if (entry.getTimestamp() != null) {
+      // timestamp
+      cells.add(new com.google.cloud.teleport.bigtable.BigtableCell(
+          BigtableUtils.bigtableRowColumnFamilyName,
+          ChangelogColumns.TIMESTAMP.getColumnNameAsByteBuffer(),
+          entry.getTimestamp(),
+          getByteBufferFromString(String.valueOf(entry.getTimestamp()), charset)
+      ));
+    }
+
+    if (entry.getTimestampFrom() != null) {
+      // timestamp_from
+      cells.add(new com.google.cloud.teleport.bigtable.BigtableCell(
+          BigtableUtils.bigtableRowColumnFamilyName,
+          ChangelogColumns.TIMESTAMP_FROM.getColumnNameAsByteBuffer(),
+          entry.getTimestamp(),
+          getByteBufferFromString(String.valueOf(entry.getTimestampFrom()), charset)
+      ));
+    }
+
+    if (entry.getTimestampTo() != null) {
+      // timestamp_to
+      cells.add(new com.google.cloud.teleport.bigtable.BigtableCell(
+          BigtableUtils.bigtableRowColumnFamilyName,
+          ChangelogColumns.TIMESTAMP_TO.getColumnNameAsByteBuffer(),
+          entry.getTimestamp(),
+          getByteBufferFromString(String.valueOf(entry.getTimestampTo()), charset)
+      ));
+    }
+
+    if (entry.getValue() != null) {
+      // value
+      cells.add(new com.google.cloud.teleport.bigtable.BigtableCell(
+          BigtableUtils.bigtableRowColumnFamilyName,
+          ChangelogColumns.TIMESTAMP_FROM.getColumnNameAsByteBuffer(),
+          entry.getTimestamp(),
+          getByteBufferFromString(String.valueOf(entry.getValue()), charset)
+      ));
+    }
+
+    return new BigtableRow(
+        createChangelogRowKey(entry.getCommitTimestamp(), workerId, counter, charset),
+        cells
+    );
+  }
+
+  private static ByteBuffer getByteBufferFromString(String s, Charset charset) {
+    return ByteBuffer.wrap(s.getBytes(charset));
+  }
+
+  private static ByteBuffer createChangelogRowKey(
+      Long commitTimestamp,
+      String workerId,
+      AtomicLong counter,
```

Comment:
I think it would look better if we pass Long here. And the incrementAndGet would be done outside

Reply:
done, moved the increment to the transform.